src/simulations/data_collector.jl

```diff
@@ -24,19 +24,24 @@ function data_collecter_aggregate(model::ABM, field_aggregator::Dict; step=1)
   end
   output = Array{Any}(undef, ncols)
   output[1] = step
-  agentslen = nagents(model)
+  agent_ids = keys(model.agents)
   counter = 2
+  rand_agent_id = 0
+  for aa in agent_ids
+    rand_agent_id = aa 
+    break
+  end
   for (fn, aggs) in field_aggregator
```

Comment:
A quick comment, not for this PR but for something to think about. What happens in this loop is very type-unstable, and thus unperformant.

Reply:
Do you mean the `temparray` is type unstable?

Reply:
Yeap. (but again, unrelated to this PR, this is something to consider after merging)